src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java

```diff
@@ -87,6 +91,14 @@ public boolean validCheckForObject(final AtlasObject object)
     protected Optional<CheckFlag> flag(final AtlasObject object)
     {
         final Set<String> instructions = new HashSet<>();
+
+        // Skip if roundabout contains synthetic node. bug fix:
+        // https://github.com/osmlab/atlas-checks/issues/316
+        if (this.isEdgeWithSyntheticBoundaryNode(object))
+        {
+            return Optional.empty();
```

Comment:
nit: could `markAsFlagged()` the object and the other edges picked up here to avoid reprocessing them